File: app/src/main/java/art/manguste/android/ArasakaWeapons/data/ProductInOrder.java
package art.manguste.android.ArasakaWeapons.data;

/**
 * Class for the product in order. Contains extra data, e.g. items count.
 */
public class ProductInOrder{
    private Product product;
    private int itemsInOrder; // how many items of this product in order

    public ProductInOrder(Product product, Integer itemsCount) {
        this.product = product;
        itemsInOrder = itemsCount > 0 ? itemsCount : 1;
    }

    public int getItemsInOrder() {
        return itemsInOrder;
    }

    public void setItemsInOrder(int itemsInOrder) {
        this.itemsInOrder = itemsInOrder;
        checkNumInRange();
    }

    public void changeItemsInOrder(int itemsInOrder) {
        this.itemsInOrder += itemsInOrder;
        checkNumInRange();
    }

    /**
     * Control items count
     */
    private void checkNumInRange(){
        if (itemsInOrder > Order.getMaxNumPerProduct()){
            this.itemsInOrder = Order.getMaxNumPerProduct();
        } else if (itemsInOrder < Order.getMinNumPerProduct()){
            this.itemsInOrder = Order.getMinNumPerProduct();
        }
    }

    public Product getProduct() {
        return product;
    }
}

File: app/src/main/java/art/manguste/android/arasakaWeapons/data/ProductInOrder.kt
package art.manguste.android.arasakaWeapons.data

/**
 * Class for the product in order. Contains extra data, e.g. items count.
 */
class ProductInOrder(val product: Product, itemsCount: Int) {
    // how many items of this product in order
    var itemsInOrder : Int = if (itemsCount > 0) itemsCount else 1
        set(value) {
            field = when {
                value > Order.maxNumPerProduct -> {
                    Order.maxNumPerProduct
                }
                value < Order.minNumPerProduct -> {
                    Order.minNumPerProduct
                }
                else -> value
            }
        }

}